### ISSUE TITLE
Support installation with mint

### ISSUE THREAD

**Issue by joshfriend** [Flinesoft/BartyCrouch#106]
https://github.com/yonaskolb/mint

the SPM build currently fails with a bunch of errors like the following:

```
BartyCrouch/Sources/Code/CommandLineParser.swift:22:45: error: use of undeclared type 'StringOption'
    public typealias CommonOptions = (path: StringOption, override: BoolOption, verbose: BoolOption)
                                            ^~~~~~~~~~~~
```

**Comment by Dschee** [comment]
Thank you for reporting this issue. I plan to rewrite some of the main parts of BartyCrouch very soon and will also make sure it installs perfectly fine via SPM (which Mint uses) – so this issue should be solved along the way. If anybody has a quick fix, feel free to post a PR – other than that, please be patient a little until I release the cleaned up version of BartyCrouch.

**Comment by iStefo** [comment]
For anyone wondering about the current progress:

When running `mint install Flinesoft/BartyCrouch@master`, the install process fails because of two reasons (as far as I can tell):

- Dschee/Polyglot does not have a `Package.swift`
- jatoben/CommandLine has a `Package.swift`, but from Swift 3 which is not compatible

The other dependencies should be fine to install via SPM/mint.

With CommandLine, it would probably be better to use SPM's [Utility.ArgumentParser](https://github.com/apple/swift-package-manager/blob/master/Sources/Utility/ArgumentParser.swift) because it is guaranteed to be maintained and future-proof.

As for Polyglot, since this is @Dschee's framework, maybe you can weight in what to do? I guess adding a `Package.swift` would be easy enough.

**Comment by Dschee** [comment]
@iStefo As you have already noticed in your update, there's version 4.0 on the way with a big refactoring where also the install process will change from Homebrew to SwiftPM. Specifically here are the features that will be included in version 4 (the checkmarked are already implemented):

- [x] Recommended [installation option](https://github.com/Flinesoft/BartyCrouch/tree/work/big-refactoring#installation) is via Mint (whilst other options remain)
- [x] Use [configuration file](https://github.com/Flinesoft/BartyCrouch/tree/work/big-refactoring#configuration) instead of thousands of command line options
- [x] [Own client implementation](https://github.com/Flinesoft/BartyCrouch/tree/work/big-refactoring/Sources/BartyCrouchTranslator) of updated Microsowft Translator API
- [ ] [Demo project based](https://github.com/Flinesoft/BartyCrouch/tree/work/big-refactoring/Demo/Untouched) integration tests (work in progress)
- [ ] Sophisticated SwiftGen-Integration (automatic static NSLocalizedString code replacement)
- [ ] More resilient search behavior (to fix issues such as #64, #87, #102, #105)

Unfortunately the update took much longer than I wanted, I just had a lot of other work to do recently. But I'm back on track since this week and plan to finish the release around next Friday. Stay tuned.

**Comment by Dschee** [comment]
This was fixed via #114.
